engine/sdk_tx_getter.go

```diff
@@ -0,0 +1,51 @@
+package engine
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"iter"
+)
+
+type sdkTxGetter struct {
+	client *Client
+}
+
+func newSDKTxGetter(client *Client) *sdkTxGetter {
+	return &sdkTxGetter{client: client}
+}
+
+func (g *sdkTxGetter) GetTransactions(ctx context.Context, ids iter.Seq[string]) ([]*sdk.Transaction, error) {
+	db := g.client.Datastore().DB()
+
+	queryIDsCtx, cancel := context.WithTimeout(ctx, 20*time.Second)
```

Comment:
```suggestion
	queryIDsCtx, cancel := context.WithTimeout(ctx, 20*time.Second)
```
the context is good for preventing long-running queries, but you might want to consider making this configurable, or at lease to move it to constant var